support/util/util_test.go

```diff
@@ -485,3 +464,279 @@ func TestSanitizeIgnitionPayload(t *testing.T) {
+			name: "HC has right pull secret name; pull secret missing key; err",
+			hc: &hyperv1.HostedCluster{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "hc",
+					Namespace: "test",
+				},
+				Spec: hyperv1.HostedClusterSpec{
+					PullSecret: corev1.LocalObjectReference{Name: "pull-secret"},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pull-secret",
+					Namespace: "test",
+				},
+				Data: map[string][]byte{
+					"wrong-key": []byte("mysecret"),
+				},
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			objs := []crclient.Object{tc.hc, tc.secret}
+
+			client := fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(objs...).Build()
+
+			pullSecretBytes, err := GetPullSecretBytes(context.TODO(), client, tc.hc)
+			if !tc.expectErr {
+				g.Expect(err).To(BeNil())
+				g.Expect(pullSecretBytes).To(Equal(tc.secret.Data[corev1.DockerConfigJsonKey]))
+			} else {
+				g.Expect(err).To(HaveOccurred())
+			}
+		})
+	}
+}
+
+func TestGetImageArchitecture(t *testing.T) {
+	pullSecretBytes := []byte("{\"auths\":{\"quay.io\":{\"auth\":\"\",\"email\":\"\"}}}")
+
+	testCases := []struct {
+		name                  string
+		image                 string
+		pullSecretBytes       []byte
+		imageMetadataProvider *RegistryClientImageMetadataProvider
+		expectedArch          hyperv1.PayloadArchType
+		expectErr             bool
+	}{
+		{
+			name:                  "Get amd64 from amd64 image; no err",
+			image:                 "quay.io/openshift-release-dev/ocp-release:4.16.10-x86_64",
+			pullSecretBytes:       pullSecretBytes,
+			imageMetadataProvider: &RegistryClientImageMetadataProvider{},
+			expectedArch:          hyperv1.AMD64,
+			expectErr:             false,
+		},
+		{
+			name:                  "Get ppc64le from ppc64le image; no err",
+			image:                 "quay.io/openshift-release-dev/ocp-release:4.16.11-ppc64le",
+			pullSecretBytes:       pullSecretBytes,
+			imageMetadataProvider: &RegistryClientImageMetadataProvider{},
+			expectedArch:          hyperv1.PPC64LE,
+			expectErr:             false,
+		},
+		{
+			name:                  "Bad pull secret; err",
+			image:                 "quay.io/openshift-release-dev/ocp-release:4.16.11-ppc64le",
+			pullSecretBytes:       []byte(""),
+			imageMetadataProvider: &RegistryClientImageMetadataProvider{},
+			expectedArch:          "",
+			expectErr:             true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			arch, err := getImageArchitecture(context.TODO(), tc.image, tc.pullSecretBytes, tc.imageMetadataProvider)
+			if tc.expectErr {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).To(BeNil())
+				g.Expect(arch).To(Equal(tc.expectedArch))
+			}
+		})
+	}
+}
+
+func TestDetermineHostedClusterPayloadType(t *testing.T) {
```

Comment:
typo on this test name TestDetermineHostedClusterPayloadType/ TestDetermineHostedClusterPayloadArch/